### PR TITLE
Update linodes field in subnet object and add active field in interface object

### PR DESCRIPTION
**Pull request by amisiorek-akamai**
## 📝 Description

Updates linodes field in a VPC subnet object from a list of Linode IDs to a list of Linode objects and adds a new field named active in the Linode config interface return object. Not yet functional as nested JSON is not being processed correctly.

## ✔️ How to Test

`make testunit`

### CODE REVIEW

File: linode_api4/objects/vpc.py [hunk 1]
```diff
@@ -1,7 +1,37 @@
...
+        interfaces = []
+
+        for c in self.interfaces:
+            if isinstance(c, VPCSubnetLinodeInterface):
+                interfaces.append(c.dict)
+            else:
+                interfaces.append(c)
+
+        partial["interfaces"] = interfaces
+        return partial
```
Comment (zliang-akamai) [inline linode_api4/objects/vpc.py:34]: This method is not needed as we used [`asdict`](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) in the base class.
```suggestion
```

File: linode_api4/objects/vpc.py [hunk 2]
```diff
@@ -24,25 +54,18 @@ class VPCSubnet(DerivedBase):
         "updated": Property(is_datetime=True),
     }
 
-    def _populate(self, json):
+    def _serialize(self):
         """
```
Comment (zliang-akamai) [inline linode_api4/objects/vpc.py:58]: I think we can keep the older `populate` method because it is functioning differently from `_serialize`.

**Comment by zliang-akamai** [comment]
We can change `"linodes": Property(),` to be `"linodes": Property(json_object=VPCSubnetLinode),` and it should be working fine automatically without any implementation of `_serialize`.

File: linode_api4/objects/vpc.py (hunk 2 above)
Comment (amisiorek-akamai) [reply linode_api4/objects/vpc.py:58]: It looks like it's working without the populate method?

**Comment by amisiorek-akamai** [comment]
I tried to do `"linodes": Property(json_object=VPCSubnetLinode),` but it was not functional for the VPCSubnetLinodeInterface component. It was struggling to serialize the object.

File: linode_api4/objects/serializable.py [hunk 3]
```diff
@@ -32,19 +38,20 @@ def from_json(cls, json: Dict[str, Any]) -> "JSONObject":
...
+                    setattr(
+                        obj,
+                        k,
+                        [
+                            cls._try_from_json(item, list_item_type)
+                            for item in json_value
+                        ],
+                    )
+                    continue
+            setattr(obj, k, cls._try_from_json(json_value, field_type))
```
Comment (lgarber-akamai) [inline linode_api4/objects/serializable.py:54]: Optional:
For readability you might be able to break this out into a few functions:

```python
    @classmethod
    def _parse_attr_list(cls, json_value, field_type):
        """
        Attempts to parse a list attribute with a given value and field type.
        """

        type_hint_args = get_args(field_type)

        if len(type_hint_args) < 1:
            return cls._try_from_json(json_value, field_type)

        return [
            cls._try_from_json(item, type_hint_args[0]) for item in json_value
        ]

    @classmethod
    def _parse_attr(cls, json_value, field_type):
        """
        Attempts to parse an attribute with a given value and field type.
        """

        if list in (field_type, get_origin(field_type)):
            return cls._parse_attr_list(json_value, field_type)

        return cls._try_from_json(json_value, field_type)

    @classmethod
    def from_json(cls, json: Dict[str, Any]) -> "JSONObject":
        """
        Creates an instance of this class from a JSON dict.
        """
        obj = cls()

        type_hints = get_type_hints(cls)

        for k in vars(obj):
            setattr(obj, k, cls._parse_attr(json.get(k), type_hints.get(k)))

        return obj
```

File: linode_api4/objects/vpc.py [hunk 4]
```diff
@@ -19,31 +33,11 @@ class VPCSubnet(DerivedBase):
         "id": Property(identifier=True),
         "label": Property(mutable=True),
         "ipv4": Property(),
-        "linodes": Property(),
+        "linodes": Property(json_object=VPCSubnetLinode),
```
Comment (lgarber-akamai) [inline linode_api4/objects/vpc.py:36]: Side note: I wonder if there's a way we could explicitly specify that this should be a list value.

**Review by lgarber-akamai — APPROVED** [review]
All tests pass locally and the implementation looks good to me. Great work!!

File: linode_api4/objects/serializable.py (hunk 3 above)
Comment (amisiorek-akamai) [reply linode_api4/objects/serializable.py:54]: I'll do that now, good idea!

**Review by zliang-akamai — APPROVED** [review]

**Comment by zliang-akamai** [comment]
Manually tested and it worked well!

**Pull request merged**